### PR TITLE
Boris: alltestcases pass

### PR DESCRIPTION
**Pull request by closerearth**

### CODE REVIEW

File: src/Crowdsale/DynamicPriceSale.sol [hunk 1]
```diff
@@ -70,8 +70,10 @@ contract DynamicSale is ContextUpgradeable, ReentrancyGuardUpgradeable, Ownable2
         quote = IERC20Upgradeable(quote_);
         minter = IMinterDAO(minter_);
         currentPrice = 222 ether;
-        saleHardCap = 7000 ether;
+        saleHardCap = 70000 ether;
         treasury = treasury_;
+        priceCurveMinValue = 4109 ether;
+        priceCurveMaxValue = 200000 ether;
```
Comment (closerearth) [inline src/Crowdsale/DynamicPriceSale.sol:76]: Why set these values here? The values should match what's in the spreadsheet: https://docs.google.com/spreadsheets/d/17PEOgTeItXcquVjt1hvyfUIXu9y0ZHjsAsUpS3DhWFA/edit#gid=1362743190

The idea is that as soon as we deploy the contract, then it becomes the new source of truth for these numbers.

File: src/Crowdsale/DynamicPriceSale.sol [hunk 2]
```diff
@@ -27,9 +27,9 @@ contract DynamicSale is ContextUpgradeable, ReentrancyGuardUpgradeable, Ownable2
     uint256 public saleHardCap;
     address public treasury;
     /// @dev mininum value for usable part of the curve
-    uint256 public priceCurveMinValue = 4109 ether;
```
Comment (arturictus) [inline src/Crowdsale/DynamicPriceSale.sol:30]: You can not add constants in an upgradable contract: https://docs.openzeppelin.com/upgrades-plugins/1.x/writing-upgradeable#avoid-initial-values-in-field-declarations

File: src/Crowdsale/DynamicPriceSale.sol [hunk 3]
```diff
@@ -70,8 +70,10 @@ contract DynamicSale is ContextUpgradeable, ReentrancyGuardUpgradeable, Ownable2
         quote = IERC20Upgradeable(quote_);
         minter = IMinterDAO(minter_);
         currentPrice = 222 ether;
-        saleHardCap = 7000 ether;
```
Comment (arturictus) [inline src/Crowdsale/DynamicPriceSale.sol:73]: Why this change? Does the whitepaper specify this saleHardCap?

**Review by arturictus — CHANGES_REQUESTED** [review]
If I'm not wrong, the contract was doing what it was supposed to do, but we did not have enough tests to verify that the selling price curve was mirroring the white paper specs.
The action plan should be:
- get excel with price points with the whitepaper formula
- test the contract to buy at those supply points and verify that the price matches
- if not: change the contract accordingly

File: src/Crowdsale/DynamicPriceSale.sol (hunk 2 above)
Comment (bpolania) [reply src/Crowdsale/DynamicPriceSale.sol:30]: What they recommend is not to set constants in the constructor or as a contract constant they won't be upgrdded, i.e., like this:

```
contract MyContract {
    uint256 public hasInitialValue = 42; // equivalent to setting in the constructor
}
```

Instead, they recommend doing it in the initializer which is the current case in this contract

File: src/Crowdsale/DynamicPriceSale.sol (hunk 3 above)
Comment (bpolania) [reply src/Crowdsale/DynamicPriceSale.sol:73]: Let me review this, I think I changed to test something did not change it back

**Review by arturictus — CHANGES_REQUESTED** [review]
This PR is duplicated #71, close one to understand which one we have to review
